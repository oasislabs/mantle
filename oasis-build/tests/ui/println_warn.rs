#[derive(oasis_std::Service)]
pub struct Printer;

impl Printer {
    pub fn new(ctx: &oasis_std::Context) -> Self {
        println!("hello, world!");
        Self
    }

    pub fn print(&self, ctx: &oasis_std::Context) {
        print!("sender: {:?}", ctx.sender());
        eprintln!("this is k");
        dbg!("this is also k");
    }
}

fn main() {
    oasis_std::service!(Printer);
}

fn random_function() {
    println!("don't warn");
}
