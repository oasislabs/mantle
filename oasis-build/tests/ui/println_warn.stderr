warning: `println!` writes to the service output channel. If you meant to log debugging information, use `eprintln!` or `dbg!`.
  --> $DIR/println_warn.rs:6:9
   |
6  |         println!("hello, world!");
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^
...
11 |         print!("sender: {:?}", ctx.sender());
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

